### PR TITLE
refactor: use server components

### PR DESCRIPTION
**Pull request by aarbanas**
## Description

With this PR we've completely refactored how our components are used and tried to render as much components on the server as possible. `Licences` and `Users` were refactored accordingly:
1. `page.tsx` - client component which is using `LicencesSearch` and `Licences` components
2. `LicencesSearch.tsx` - server component which is implementing new custom search hook. It propagates filter object to parent component
3. `Licences.tsx` - server component which is calling `trpc` to load the data and render the `LicencesTable`
4. `LicencesTable.tsx` - server component which is just rendering the table

Edit users by id has been divided int two components:
1. `page.tsx` - get user data using `trpc`
2. `UsersForm.tsx` - render the form with valid data

Additional features which were refactored:
1. Translated all EN labels to HR
2. On hover in menu navbar add color
3. Fix the `FormSwitch.tsx`
4. Create Pagination context so we do not have prop drilling for passing page values down to the pagination component

## Change summary
- update dependencies
- refactor licences/users page and make code more reusable

## Checklist

- [x] I have performed a self-review of my code
- [x] I tested locally
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New package or package update introduced

### CODE REVIEW

**Comment by 9helix** [comment]
It's probably debatable, but one thing that comes to mind is to combine `UsersSearch.tsx` and `LicensesSearch.tsx` into a single file that additionally has `title` and `searchKey` in the `Props` type since those attributes are the only differences in those 2 files.
Thoughts?

**Comment by aarbanas** [comment]
> It's probably debatable, but one thing that comes to mind is to combine `UsersSearch.tsx` and `LicensesSearch.tsx` into a single file that additionally has `title` and `searchKey` in the `Props` type since those attributes are the only differences in those 2 files. Thoughts?

Hey, great point. I was thinking the same thing while I was developing it. The only thing why I decided to make it this way is because in the future `UsersSearch` will have Dropdown search to filter users by city and some extra filters

**Review by 9helix — APPROVED** [review]
> Hey, great point. I was thinking the same thing while I was developing it. The only thing why I decided to make it this way is because in the future UsersSearch will have Dropdown search to filter users by city and some extra filters

Makes sense then, better to leave it like that so it is future-proofed.
Everything else seems good to me.

File: src/app/(pages)/licenses/_components/Licences.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import LicencesTable from "~/app/(pages)/licenses/_components/LicencesTable";
+import { usePagination } from "~/components/organisms/pagination/PaginationContext";
+import { api } from "~/trpc/react";
+import React, { useEffect, useState } from "react";
+import LoadingSpinner from "~/components/organisms/loadingSpinner/LoadingSpinner";
+
+type Props = {
+  filter: Record<string, string> | undefined;
+};
+const Licences: React.FC<Props> = ({ filter }) => {
```
Comment (IMilja) [inline src/app/(pages)/licenses/_components/Licences.tsx:10]: Missing new line.

File: src/app/(pages)/licenses/_components/LicencesSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
```
Comment (IMilja) [inline src/app/(pages)/licenses/_components/LicencesSearch.tsx:1]:
```suggestion
import { type FC } from "react";
```

File: src/app/(pages)/licenses/_components/LicencesSearch.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import SearchInput from "~/components/atoms/SearchInput";
+import useSearch from "~/hooks/useSearch";
+
+type Props = {
+  onSearch: (filter: Record<string, string> | undefined) => void;
+};
+
+const LicencesSearch: React.FC<Props> = ({ onSearch }) => {
```
Comment (IMilja) [inline src/app/(pages)/licenses/_components/LicencesSearch.tsx:9]:
```suggestion
const LicencesSearch: FC<Props> = ({ onSearch }) => {
```

File: src/app/(pages)/licenses/_components/LicencesTable.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import { Pencil } from "lucide-react";
+import { type FindLicenseReturnDTO } from "~/server/services/license/license.repository";
+import PaginationComponent from "~/components/organisms/pagination/PaginationComponent";
+
+type Props = {
+  data?: FindLicenseReturnDTO[];
+  totalPageNumber: number;
+};
+
+const LicencesTable: React.FC<Props> = ({ data, totalPageNumber }) => {
```
Comment (IMilja) [inline src/app/(pages)/licenses/_components/LicencesTable.tsx:18]:
```suggestion
const LicencesTable: FC<Props> = ({ data, totalPageNumber }) => {
```

File: src/app/(pages)/users/[id]/_components/UsersForm.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+      firstname: formData.firstname,
+      lastname: formData.lastname,
+    },
+  });
+
+  // Handle form submission
+  const handleSubmit = async () => {
+    setIsSubmitting(true);
+    form.setValue("active", active);
+    setIsSubmitting(false);
```
Comment (IMilja) [inline src/app/(pages)/users/[id]/_components/UsersForm.tsx:37]: Could you check if we can use the native `isSubmitting` property from the forms?
https://react-hook-form.com/docs/useform/formstate search for `isSubmitting`

File: src/app/(pages)/users/[id]/page.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
+"use client";
+import MainLayout from "~/components/layout/mainLayout";
+import { useParams } from "next/navigation";
+import { api } from "~/trpc/react";
+import LoadingSpinner from "~/components/organisms/loadingSpinner/LoadingSpinner";
+import UserForm from "~/app/(pages)/users/[id]/_components/UsersForm";
+
+export default function UserDetailPage() {
+  const { id } = useParams(); // Get the user ID from the URL
```
Comment (IMilja) [inline src/app/(pages)/users/[id]/page.tsx:9]: We can remove this comments since they bring no value.

File: src/components/layout/mainLayout.tsx
Comment (IMilja) [inline src/components/layout/mainLayout.tsx:1]: Hmm, the active routes logic works, but, I do not like the fact that we use hardcoded strings. Should we switch to an enum?

```ts
enum AppRoutesEnum {
  HOME = '/',
  EDUCATIONS = '/educations',
  ...
}
```

**Review by IMilja — CHANGES_REQUESTED** [review]
I left some comments that we can address before merging it.

File: src/app/(pages)/users/[id]/_components/UsersForm.tsx (hunk 5 above)
Comment (aarbanas) [reply src/app/(pages)/users/[id]/_components/UsersForm.tsx:37]: It is working, great catch 🚀 
image

File: src/components/layout/mainLayout.tsx [hunk 7]
```diff
@@ -39,43 +42,35 @@ const MainLayout: React.FC<Readonly<Props>> = ({
 };
 
 const Navigation = () => {
+  const currentPath = usePathname();
```
Comment (IMilja) [inline src/components/layout/mainLayout.tsx:45]:
```suggestion
  const currentPath = usePathname() as AppRoutes;
```

Could this work so you can avoid casting on all other places?

**Review by IMilja — APPROVED** [review]
+1

**Pull request merged**